provider/azure/environ.go

```diff
@@ -1074,7 +1074,56 @@ func (env *azureEnviron) instances(
 
 // AdoptResources is part of the Environ interface.
 func (env *azureEnviron) AdoptResources(controllerUUID string, fromVersion version.Number) error {
-	return errors.NotImplementedf("AdoptResources")
+	groupClient := resources.GroupsClient{env.resources}
+	resourceClient := resources.Client{env.resources}
+	var failed []string
+
+	resources, err := groupClient.ListResources(env.resourceGroup, "", nil)
+	if err != nil {
+		return errors.Annotate(err, "listing resources")
+	}
+	for resources.Value != nil {
+		for _, resource := range *resources.Value {
+
+			providerNamespace, parentPath, err := splitResourceType(*resource.Type)
```

Comment:
This is the bit I'm least certain about - from looking at the places constructing Resources in the same file this seems to be what the Type attribute is. It generally looks like "Microsoft.Compute/availabilitySets" - so this would have the provider namespace "Microsoft.Compute" and parent path "availabilitySets". 
But I don't really get why it would be needed given that we're also passing the type as another field? (And in the resource as well.) So maybe I'm misunderstanding? Or is the redundancy just an artifact of API autogeneration?